plugins/powerpc/powerpc_branch.ml

```diff
@@ -0,0 +1,277 @@
+open Powerpc.Std
+
+let update_link_register cpu ops =
+  RTL.[cpu.lr := cpu.pc + unsigned const byte 4]
+
+(** Branch Instructions, Branch
+    Page 37 of IBM Power ISATM Version 3.0 B
+    examples:
+    4b ff fe f0  b .+67108592
+    4b ff fe f2  ba 67108592
+    4b ff fe f1  bl .+67108592
+    4b ff fe f3  bla 67108592 *)
+let b cpu ops =
+  let im = signed imm ops.(0) in
+  let tm = signed var (bitwidth 26) in
+  let sh = unsigned of_string "0b00" in
+  RTL.[
+    tm := im ^ sh;
+    cpu.jmp (cpu.pc + tm)
+  ]
+
+let ba cpu ops =
+  let im = signed imm ops.(0) in
+  let tm = signed var (bitwidth 26) in
+  let sh = unsigned of_string "0b00" in
+  RTL.[
+    tm := im ^ sh;
+    cpu.jmp tm;
+  ]
+
+let bl cpu ops =
+  let im = signed imm ops.(0) in
+  let tm = signed var (bitwidth 26) in
+  let sh = unsigned of_string "0b00" in
+  let ad = unsigned const byte 4 in
+  RTL.[
+    tm := im ^ sh;
+    cpu.jmp (cpu.pc + tm);
+    cpu.lr := cpu.pc + ad;
```

Comment:
you can't update the link register after the jump is already executed. The jump takes effect immediately.